plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/TokenFetchingMetastoreClientFactory.java

```diff
@@ -70,15 +73,25 @@ public ThriftMetastoreClient createMetastoreClientFor(Optional<ConnectorIdentity
         String username = identity.map(ConnectorIdentity::getUser)
                 .orElseThrow(() -> new IllegalStateException("End-user name should exist when metastore impersonation is enabled"));
 
-        String delegationToken;
         try {
-            delegationToken = delegationTokenCache.getUnchecked(username);
+            return clientProvider.createMetastoreClient(Optional.of(getDelegationToken(username)));
+        }
+        catch (TException e) {
+            log.debug(e, "It may be a delegation token permission failure, so try again with a new delegation token");
+            delegationTokenCache.put(username, loadDelegationToken(username));
+            return clientProvider.createMetastoreClient(Optional.of(getDelegationToken(username)));
```

Comment:
you can store the delegation token obtained from `loadDelegationToken(username))` on variable; put it in the cache and then use it here without going through cache

Reply:
Thank for giving feedbacks :)
changed it to be reused as a local variable.